http2_general/test_h2_headers.py

```diff
@@ -1586,6 +1587,373 @@ def test_pragma_from_hpack_table(self):
         client.send_request(request, "200")
         self.assertEqual(3, len(server.requests))
 
+    def __reload_tempesta_with_ja5h(self, ja5_config):
+        tempesta: Tempesta = self.get_tempesta()
+        tempesta.config.defconfig += ja5_config
+        tempesta.reload()
+
+    def test_referer_from_hpack_table(self):
+        self.start_all_services()
+        client = self.get_client("deproxy")
+        server = self.get_server("deproxy")
+
+        request_hashed = client.create_request(
+            method="GET",
+            headers=[("referer", "http://tempesta-tech.com:8080")],
+        )
+        client.send_request(request_hashed, "200")
+
+        # Do not allow requests with same hash from the client.
+        self.__reload_tempesta_with_ja5h(
+            """
+            ja5h {
+                hash 1032008c02c0 0 0;
+            }
+        """
+        )
+
+        # Referer is false allow request
+        request = client.create_request(method="GET", headers=[])
+        client.send_request(request, "200")
+        self.assertEqual(2, len(server.requests))
+
+        # Request with same hash is blocked
+        client.send_request(request_hashed, "403")
+        self.assertEqual(2, len(server.requests))
+
+    def __send_add_check_req_with_huffman(self, client, request, huffman, expected_status_code):
+        # create stream and change state machine in H2Connection object
+        stream = client.init_stream_for_send(client.stream_id)
+
+        hf = frame.HeadersFrame(
+            stream_id=client.stream_id,
+            data=client.h2_connection.encoder.encode(request, huffman=huffman),
+            flags=["END_HEADERS", "END_STREAM"],
+        )
+        client.send_bytes(data=hf.serialize(), expect_response=True)
```

Comment:
Why do you use `send_bytes` method? You send a simple response with a lot of valid headers or not? I think you can use `send_request` method here/